common/src/main/java/tachyon/heartbeat/SleepingTimer.java

```diff
@@ -47,14 +47,13 @@ public SleepingTimer(String threadName, long intervalMs) {
    * @throws InterruptedException if the thread is interrupted while waiting
    */
   public void tick() throws InterruptedException {
-    long currentTickMs = System.currentTimeMillis();
-    long executionTimeMs = currentTickMs - mPreviousTickMs;
-    if (executionTimeMs > mIntervalMs) {
+    long executionTimeMs = System.currentTimeMillis() - mPreviousTickMs;
+    if (mPreviousTickMs != 0 && executionTimeMs > mIntervalMs) {
       LOG.warn(mThreadName + " last execution took " + executionTimeMs
           + " ms. Longer than the interval " + mIntervalMs);
     } else {
       Thread.sleep(mIntervalMs - executionTimeMs);
     }
-    mPreviousTickMs = currentTickMs;
+    mPreviousTickMs = System.currentTimeMillis();
```

Comment:
@jsimsa 
I also just hit on this bug (opened Tachyon-1128).
For this solution, will you try to sleep some negative ms on the first tick?